[Security][Gap Fill][9.3 & Serverless]: Doc auto gap fill feature (#4671)
elastic/docs-content: solutions/security/detect-and-alert/monitor-rule-executions.md

* **Last Gap (if any)**: Shows how long the most recent gap for a particular rule lasted.
* **Unfilled gaps duration**: Shows whether a rule still has gaps and provides a total sum of the remaining unfilled or partially filled gaps. The total sum can change based on the time range that you select during the past 90 days (data on gaps that occurred past the last 90 days is not retained). If a rule has no gaps, the columns display a dash (`––`). 

> * {applies_to}`stack: ga 9.3+``**Gap fill status**: Shows the status of the rule's gaps. If any gaps are unfilled, the gap status is `Unfilled`. If any gaps are being are being filled, the status is `In progress`. If all gaps are filled, the status is `Filled`. 

Author: ```suggestion
* {applies_to}`stack: ga 9.3+` **Gap fill status**: Shows the status of the rule's gaps. If any gaps are unfilled, the gap status is `Unfilled`. If any gaps are being are being filled, the status is `In progress`. If all gaps are filled, the status is `Filled`. 
```

    ::::

#### Manually fill gaps [rule-monitoring-tab-manually-fill-gaps]

Manually fill gaps for a specific rule by using the Gaps table on the rule's **Execution results** tab. Refer to [](#gaps-table) to learn more.

Manually fill gaps for multiple rules by using the **Fill gaps** bulk action in the Rules table. Refer to [](/solutions/security/detect-and-alert/manage-detection-rules.md#bulk-fill-gaps-multiple-rules) to learn more.

#### Automatically fill gaps [rule-monitoring-tab-auto-fill-gaps]

```{applies_to}
stack: ga 9.3+
```

::::{note}
You must have the appropriate subscription to use the automatic gap fill feature. Refer to the subscription page for [Elastic Cloud](https://www.elastic.co/subscriptions/cloud) and [Elastic Stack/self-managed](https://www.elastic.co/subscriptions) for the breakdown of available features and their associated subscription tiers.
::::

To automate gap remediation for your rules, enable the automatic gap fill feature. When turned on, a job runs every five minutes to check for new and existing gaps, then schedules tasks to fill them. Details about the job and scheduled gap fill tasks are captured in the gap fill scheduler logs. 

> You can find the setting that controls the automatic gap fill feature by clicking **Settings** (above the Rules table), then going to the **Auto gap fill settings** section. Use the toggle to turn the feature on and off. 

Reviewer: Maybe move this sentence after the first sentence in line 100? I think then the line about gap fill tasks being captured in the scheduler logs  could transition nicely into the line about how to access the logs and log details.

Author: Fixed in [`2fe6e30` (#4671)](https://github.com/elastic/docs-content/pull/4671/commits/2fe6e30751acb207282ab1973afc5f9dea91273a).

    ::::{tip}
    :applies_to:{stack: ga 9.3+}
    Use the **Gap fill status** filter in the Rules table to find rules with the specified gap status. 
    ::::

#### Manually fill gaps [rule-monitoring-tab-manually-fill-gaps]

You can manually fill gaps in the following ways:

- **For a specific rule**: Use the **Fill gaps** action in the rule's Gaps table, which lets you assess and respond to existing gaps. Refer to [](#gaps-table) to learn more.

- **For multiple rules**: Use the **Fill gaps** bulk action in the Rules table. Refer to [](/solutions/security/detect-and-alert/manage-detection-rules.md#bulk-fill-gaps-multiple-rules) to learn more.

#### Automatically fill gaps [rule-monitoring-tab-auto-fill-gaps]

```{applies_to}
stack: ga 9.3+
```

::::{note}
You must have the appropriate subscription to use the automatic gap fill feature. Refer to the subscription page for [Elastic Cloud](https://www.elastic.co/subscriptions/cloud) and [Elastic Stack/self-managed](https://www.elastic.co/subscriptions) for the breakdown of available features and their associated subscription tiers.
::::

> To automate gap remediation for your rules, enable the automatic gap fill feature. When turned on, a job runs every five minutes to check for new and existing gaps, then schedules tasks to fill them. 

Reviewer 2: Yes

You can find the setting that controls the automatic gap fill feature by clicking **Settings** (above the Rules table), then going to the **Auto gap fill settings** section. Use the toggle to turn the feature on and off. 

::::{tip}
The **Auto gap fill status:** field (located in the panel above the Rules table) shows whether the automatic gap fill feature is on or off. Click the field value to access the automatic gap fill settings.
::::

> Details about the job and scheduled gap fill tasks are captured in the gap fill scheduler logs. You access the gap fill scheduler logs by clicking **logs** in the **Auto gap fill settings** section description. 

Reviewer 3: by clicking "Gap fill scheduler"

In the scheduler logs table, expand the rows to learn more about gaps that were discovered and gap fill tasks that were scheduled each time the automatic gap fill job ran. You can also find key details such as:

- When each job ran to check for gaps
- The number of gaps each job detected and the number of rules that had gaps
- The number of gap fill tasks each job scheduled 
- The status of the scheduled gap fill tasks. They can be:

>     - **Success**: Confirms that scheduled gap fill tasks were sucessfully run for all rules with gaps.

Reviewer 2: Is not necessary **all** rules with gams.

Let's say we have 1000 rules with gaps, in the task execution we only process 300, and task is timeouted.

So, we will report it's as success, and in later execution process rules which left.

Author: ```suggestion
    - **Success**: Confirms that gap fill tasks were successfully scheduled for rules assessed by the auto gap fill job. Refer to the log details to learn if any rules were not processed. 
```

>     - **Success**: Confirms that scheduled gap fill tasks were sucessfully run for all rules with gaps.

Author: Fixed in https://github.com/elastic/docs-content/pull/4671#discussion_r2710516457 : )

>     - **Error**: Indicates that scheduled gap fill tasks failed to run.    

Reviewer 2: We can also have **error** status when we schedule gap fills for some rules, but at least 1 gap fill schedule is failed. 

So there can be some successful, but at least 1 failed gap fils schedule.

Author: ```suggestion
    - **Error**: Indicates that some gap fill tasks were not scheduled or scheduled gap fill tasks failed to run. Refer to the log details for more information.
```
